### PR TITLE
canu: import from brewsci/science

### PR DESCRIPTION
**Pull request by emreerhan**
- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

File: Formula/canu.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class Canu < Formula
+  desc "Single molecule sequence assembler"
+  homepage "https://canu.readthedocs.org/en/latest/"
+  url "https://github.com/marbl/canu/archive/v1.6.tar.gz"
+  sha256 "470e0ac761d69d1fecab85da810a6474b1e2387d7124290a0e4124d660766498"
+  head "https://github.com/marbl/canu.git"
+  # cite "10.1038/nbt.3238"
```
Comment (sjackman) [inline Formula/canu.rb]: Please add `https://doi.org/…`

File: Formula/canu.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  # Fix fatal error: 'omp.h' file not found
+  depends_on "gcc"
```
Comment (sjackman) [inline Formula/canu.rb]: Please add
```sh
depends_on "gcc" unless OS.mac? # for openmp
```

File: Formula/canu.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  def install
+    system "make", "-C", "src"
+    arch = Pathname.new(Dir["*/bin"][0]).dirname
+    rm_r arch/"obj"
+    prefix.install arch
+    bin.install_symlink prefix/arch/"bin/canu"
+    doc.install Dir["README.*"]
```
Comment (sjackman) [inline Formula/canu.rb]: You can remove this line. It's done automatically now.

File: Formula/canu.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  test do
+    system "#{bin}/canu", "--version"
```
Comment (sjackman) [inline Formula/canu.rb]: Please modify this test to use `assert_match` and `shell_output`. You can see examples in other formulae.

File: Formula/canu.rb (hunk 3 above)
Comment (emreerhan) [reply Formula/canu.rb]: Which line?

File: Formula/canu.rb [hunk 5]
```diff
@@ -11,7 +11,6 @@ class Canu < Formula
 
   def install
     system "make", "-C", "src"
-    arch = Pathname.new(Dir["*/bin"][0]).dirname
```
Comment (sjackman) [inline Formula/canu.rb]: I think you misinterpreted my previous comment. I only meant delete the last line. Sorry for the confusion
Delete this:
```ruby
    doc.install Dir["README.*"]
```
Keep this:
```ruby
    arch = Pathname.new(Dir["*/bin"][0]).dirname
    rm_r arch/"obj"
    prefix.install arch
    bin.install_symlink prefix/arch/"bin/canu"
```

File: Formula/canu.rb [hunk 6]
```diff
@@ -10,7 +10,12 @@ class Canu < Formula
   depends_on "gcc" unless OS.mac?
 
   def install
-    system "make", "-C", "src"
+    system "make", "-C", "src" 
+    # bin.install "canu"
```
Comment (sjackman) [inline Formula/canu.rb]: Please remove this comment.

**Comment by sjackman** [comment]
Please squash the four commits down to a single commit and rebase onto `origin/master`. You can use `git rebase -i origin/master` for this purpose.

File: Formula/canu.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
+class Canu < Formula
+  desc "Single molecule sequence assembler"
+  homepage "https://canu.readthedocs.org/en/latest/"
+  url "https://github.com/marbl/canu/archive/v1.6.tar.gz"
+  sha256 "470e0ac761d69d1fecab85da810a6474b1e2387d7124290a0e4124d660766498"
+  head "https://github.com/marbl/canu.git"
+  # cite "doi.org/10.1101/gr.215087.116"
```
Comment (sjackman) [inline Formula/canu.rb]: See https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md#citation
(I wrote this text just now.)

File: Formula/canu.rb [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+  # Fix fatal error: 'omp.h' file not found
+  depends_on "gcc" unless OS.mac?
```
Comment (sjackman) [inline Formula/canu.rb]: Please add the comment
```ruby
depends_on "gcc" unless OS.mac? # for openmp
```

**Comment by emreerhan** [comment]
`brew test brewsci/bio/canu` is failing for some reason. I can execute the binary without problems manually, but it fails when I use `brew test`:

`Error: brewsci/bio/canu: failed
</in\.paf/> expected to be =~`

Any ideas?

**Comment by sjackman** [comment]
The test is running the command `canu` and looking for the string `in.paf` in its output.

File: Formula/canu.rb [hunk 9]
```diff
@@ -0,0 +1,23 @@
+class Canu < Formula
+  # cite Koren_2017 "doi.org/10.1101/gr.215087.116"
```
Comment (sjackman) [inline Formula/canu.rb]: Add a colon `:` after Koren_2017. Add `https://` before `doi.org`

**Comment by sjackman** [comment]
```
Error: 1 problem in 1 formula
brewsci/bio/canu:
  * C: 13: col 31: Trailing whitespace detected.
==> FAILED
```

**Comment by sjackman** [comment]
Please squash the commits down to a single commit and rebase onto origin/master. You can use git rebase -i origin/master for this purpose.

**Comment by sjackman** [comment]
Merged. Thanks, Emre!

**Pull request closed without merging**